### PR TITLE
Support for TIMESTAMPDIFF SQL function

### PR DESCRIPTION
**Pull request by mihaibudiu**

### CODE REVIEW

**Comment by mihaibudiu** [comment]
Turns out Calcite has a bug in computing this function. I have submitted a fix: https://github.com/apache/calcite-avatica/pull/226

**Comment by mihaibudiu** [comment]
@lalithsuresh are you still reviewing this?

File: docs/sql/datetime.md [hunk 1]
```diff
@@ -148,7 +156,10 @@ Values of type `TIMESTAMP` can be compared using `=`, `<>`, `!=`, `<`,
 
 `TIMESTAMPDIFF(<unit>, left, right)` computes the difference between
 two timestamps and expresses the result in the specified time units.
-The result is a 32-bit integer.
+The result is a 32-bit integer.  `DATEDIFF` is a synonim for
```
Comment (lalithsuresh) [inline docs/sql/datetime.md:159]: synonym

File: docs/sql/datetime.md [hunk 2]
```diff
@@ -148,7 +156,10 @@ Values of type `TIMESTAMP` can be compared using `=`, `<>`, `!=`, `<`,
...
+`TIMESTAMPDIFF`.  One month is considered elapsed when the calendar
+month has increased and the calendar day and time is equal or greater
```
Comment (lalithsuresh) [inline docs/sql/datetime.md:161]: time is greater than or equal to the start.

**Review by lalithsuresh — APPROVED** [review]

**Pull request merged**